recipe/cross_compile.diff

```diff
@@ -0,0 +1,126 @@
+diff --git a/install/cupy_builder/_features.py b/install/cupy_builder/_features.py
+index d12de78c3..77c29a954 100644
+--- a/install/cupy_builder/_features.py
++++ b/install/cupy_builder/_features.py
+@@ -478,6 +478,7 @@ class CUDA_cuda(Feature):
+             ''', include_dirs=settings['include_dirs'])  # type: ignore[no-untyped-call] # NOQA
+         except Exception as e:
+             utils.print_warning('Cannot check CUDA version', str(e))
++            raise
+             return False
+ 
+         self._version = int(out)
+diff --git a/install/cupy_builder/cupy_setup_build.py b/install/cupy_builder/cupy_setup_build.py
+index 8023e46ad..a9f746d0d 100644
+--- a/install/cupy_builder/cupy_setup_build.py
++++ b/install/cupy_builder/cupy_setup_build.py
+@@ -294,16 +294,41 @@ def make_extensions(ctx: Context, compiler, use_cython):
+         # deprecated since ROCm 4.2.0
+         settings['define_macros'].append(('__HIP_PLATFORM_HCC__', '1'))
+ 
+-    available_modules = []
+-    if no_cuda:
+-        available_modules = [m['name'] for m in MODULES]
+-    else:
+-        available_modules, settings = preconfigure_modules(
+-            ctx, MODULES, compiler, settings)
+-        required_modules = get_required_modules(MODULES)
+-        if not (set(required_modules) <= set(available_modules)):
+-            raise Exception('Your CUDA environment is invalid. '
+-                            'Please check above error log.')
++    try:
++        build_compiler = compiler
++        cc = os.environ['CC']
++        #ar = os.environ['AR']
++        cflags = os.environ['CFLAGS']
++        ldflags = os.environ['LDFLAGS']
++        if os.environ.get('CONDA_BUILD_CROSS_COMPILATION'):
++            # If cross-compiling, we need build_and_run() to use the compiler
++            # on the build platform.
++            os.environ['CC'] = os.environ['CC_FOR_BUILD']
```

Comment:
I checked that this line alone is enough to convince `distutils` to use the x86-64 compiler form `compile` and `link`, so I have no idea why we'd get a linker error that involves `ld` for aarch64...

Reply:
It seems a few lines below
```python
sysconfig.customize_compiler(compiler)
```
is imposing the cross compiler setting onto the native compiler, thus causing errors.